src/hypothesis/database.py

```diff
@@ -240,7 +240,11 @@ def fetch(self, key):
     def save(self, key, value):
         path = self._value_path(key, value)
         if not os.path.exists(path):
-            tmpname = path + '.' + str(binascii.hexlify(os.urandom(16)))
+            suffix = binascii.hexlify(os.urandom(16))
+            if not isinstance(suffix, str):
```

Comment:
I think you probably want something like `if not isinstance(suffix, future.utils.text_type)` otherwise this has the opposite behavior on python2